app/code/community/Bolt/Boltpay/Model/Order.php

```diff
@@ -291,31 +302,182 @@ public function createOrder($reference, $sessionQuoteId = null, $isAjaxRequest =
+            $invalidSKUs = array_unique(array_merge($outOfStockSKUs, $outOfStockSKUs));
+            $errorMessage = Mage::helper('boltpay')->__("Please review " .
+                (count($invalidSKUs) > 1 ? "them" : "it") .
+                " and un-hold the order. ");
+
+            $this->appendOrderOnHoldMessage($errorMessage);
+        }
+    }
+
+    /**
+     * @return void
+     */
+    protected function setOrderOnHoldFlag()
+    {
+        $this->isOrderOnHold = true;
+    }
+
+    /**
+     * @return boolean
+     */
+    protected function isOrderOnHoldFlagSet()
+    {
+        return $this->isOrderOnHold;
+    }
+
+    /**
+     * @param $message
      *
-     * @var Mage_Sales_Model_Quote $quote   The quote that defines the cart
+     * @return void
+     */
+    protected function appendOrderOnHoldMessage($message)
+    {
+        $this->orderOnHoldMessage .= $message;
+    }
+
+    /**
+     * @var Mage_Sales_Model_Quote $quote The quote that defines the cart
      *
-     * @return bool true if the store can accept an order for all items in the cart,
-     *              otherwise, false
+     * @return array
      */
-    public function storeHasAllCartItems($quote)
+    public function getOutOfStockSKUs(Mage_Sales_Model_Quote $quote)
     {
-        foreach ($quote->getAllItems() as $cartItem) {
-            if($cartItem->getHasChildren()) {
-                continue;
+        if($this->outOfStockSkus == null) {
+            $this->outOfStockSkus = array();
+
+            foreach($this->getCartProducts($quote) as $product) {
+                $stockInfo = Mage::getModel('cataloginventory/stock_item')->loadByProduct($product);
+                if ($stockInfo->getManageStock()) {
+                    if (($stockInfo->getQty() < $product->getCartItemQty()) && !$stockInfo->getBackorders()) {
+                        $this->outOfStockSkus[] = $product->getSku();
+                    }
+                }
             }
+        }
 
-            $_product = Mage::getModel('catalog/product')->load($cartItem->getProductId());
-            $stockInfo = Mage::getModel('cataloginventory/stock_item')->loadByProduct($_product);
-            if($stockInfo->getManageStock()){
-                if( ($stockInfo->getQty() < $cartItem->getQty()) && !$stockInfo->getBackorders() ){
-                    return false;
+        return $this->outOfStockSkus;
+    }
+
+    /**
+     * @var Mage_Sales_Model_Quote $quote The quote that defines the cart
+     *
+     * @return array
+     */
+    protected function getCartProducts(Mage_Sales_Model_Quote $quote)
+    {
+        if($this->cartProducts == null) {
+            foreach ($quote->getAllItems() as $cartItem) {
+                if ($cartItem->getHasChildren()) {
+                    continue;
                 }
+
+                /** @var Mage_Catalog_Model_Product $product */
+                $product = Mage::getModel('catalog/product')->load($cartItem->getProductId());
```

Comment:
Can you check does it works here?

`class app/code/core/Mage/Sales/Model/Quote/Item/Abstract.php::getProduct()`

current method should load data if the data does not exist. If data existы it will load from the cache.